client/client.go

```diff
@@ -282,15 +250,27 @@ func (c *Client) withAccountIDRegionAndNamespace(accountID, region, namespace st
 	}
 }
 
+func validateRegions(logger hclog.Logger, regions []string) (bool, error) {
```

Comment:
I'd add documentation on this method as in additional to `validateRegions` it return if it's also wildcard.

Reply:
Another idea is to split it to two functions. `isValidRegions(regions []string) error` and `isAllRegions(regions []string) bool` this way it will be also more self-explanatory.

Reply:
changed